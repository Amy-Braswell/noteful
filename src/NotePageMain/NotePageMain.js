import React from 'react'
import Note from '../Note/Note'
import ApiContext from '../ApiContext'
import config from '../config'
import { findNote } from '../notes-helpers'
import './NotePageMain.css'

export default class NotePageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  deleteNote = noteId => {
    this.props.history.push(`/`)
  }

  handleClickEdit = e => {
    e.preventDefault()
    const noteId = this.props.id
    fetch(`${config.API_ENDPOINT}/notes/${noteId}`, {
      method: 'PATCH',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(() => {
        this.context.editNote(noteId)
        // allow parent to perform extra behaviour
        this.props.onEditNote(noteId)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render(){
    const { notes=[] } = this.context
    const { noteId } = this.props.match.params
    const note = findNote(notes, noteId) || { content: '' }
    return(
        <section className='NotePageMain'>
        <Note
          id={note.id}
          name={note.name}
          modified={note.modified}
          onDeleteNote={this.deleteNote}
          onEditNote={this.editNote}
        />
        <div className='NotePageMain__content'>
          {note.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )}
        </div>
      </section>
    )
  }
}