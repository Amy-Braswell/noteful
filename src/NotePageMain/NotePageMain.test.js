import React from 'react'
import renderer from 'react-test-renderer'
import NotePageMain from './NotePageMain'

describe.only(`NotePageMain component`, () => {
  const props = {
    note: {
      "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Dogs",
      "modified": "2019-01-03T00:00:00.000Z",
      "folderid": "2",
      "content": "Corporis accusamus placeat.\n \rUnde."
    },
    match: {
      params: {
         noteId: "123"
      }
    }
  }

  it('renders a .NotePageMain by default', () => {
    const tree = renderer
      .create(<NotePageMain />)
      .toJSON()
      expect(tree).toMatchSnapshot()
  })

  it('renders a Note with note prop', () => {
    const note = renderer
      .create(<NotePageMain {...props} />)
      .find('Note')
      .toJSON()
    expect(note).toMatchSnapshot()
  })

  it(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    [{
      note: { "content": "Content with n r.\n \rafter n r." }
    }, {
      note: { "content": "Content with n.\nafter." }
    }].forEach(props => {
      const content = renderer
        .create(<NotePageMain {...props} />)
        .find('NotePageMain__content')
        .toJSON()
      expect(content).toMatchSnapshot()
    })
  })
})
