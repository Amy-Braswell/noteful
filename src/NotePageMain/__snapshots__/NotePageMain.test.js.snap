// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotePageMain component renders a .NotePageMain by default 1`] = `
<section
  className="NotePageMain"
>
  <div
    className="NotePageMain__content"
  >
    <p>
      
    </p>
  </div>
</section>
`;
